### PR TITLE
Define `==` and `isapprox` for all geometries

### LINKED ISSUE #892: Add `==` and `isapprox` methods for all `Geometry` subtypes

### PR DESCRIPTION
**Pull request by eliascarv**
closes #892

### CODE REVIEW

File: src/multigeoms.jl [hunk 1]
```diff
@@ -54,7 +54,7 @@ rings(m::MultiPolygon) = [ring for poly in m.geoms for ring in rings(poly)]
 
 Base.parent(m::Multi) = m.geoms
 
-==(m₁::Multi, m₂::Multi) = length(m₁.geoms) == length(m₂.geoms) && all(g -> g[1] == g[2], zip(m₁.geoms, m₂.geoms))
+==(m₁::Multi, m₂::Multi) = m₁.geoms == m₂.geoms
```
Comment (juliohm) [inline src/multigeoms.jl:57]: is this equivalent performance-wise?

File: src/primitives/point.jl [hunk 2]
```diff
@@ -49,7 +49,7 @@ paramdim(::Type{<:Point}) = 0
 
 center(p::Point) = p
 
-==(A::Point, B::Point) = A.coords == B.coords
+==(A::Point, B::Point) = to(A) == to(B)
```
Comment (juliohm) [inline src/primitives/point.jl:52]: can you please confirm that this is as fast as before with Cartesian coords?

File: test/polytopes.jl [hunk 3]
```diff
@@ -101,6 +105,13 @@
     c3 = Ring(T.((1, 1.0)), T.((2.0, 2.0)))
     @test c1 == c2 == c3
 
+    c = Rope(point(0, 0), point(1, 0), point(0, 1))
+    @test c == Rope(Point(0.0, 0.0), Point(1.0, 0.0), Point(0.0, 1.0))
+    @test c == Rope(Point(0.0f0, 0.0f0), Point(1.0f0, 0.0f0), Point(0.0f0, 1.0f0))
```
Comment (juliohm) [inline test/polytopes.jl]:
```suggestion
    @test c == Rope(Point(0.0f0, 0.0f0), Point(1.0f0, 0.0f0), Point(0.0f0, 1.0f0))
    
```

File: test/runtests.jl [hunk 4]
```diff
@@ -87,6 +87,28 @@ randpoint2(n) = randpoint(2, n)
...
+  ctor = Meshes.constructor(G)
+  names = fieldnames(G)
+  exprs = (:(withprecision(T, g.$name)) for name in names)
+  :($ctor($(exprs...)))
+end
+
+function equaltest(g)
+  @test g == withprecision(Float64, g)
+  @test g == withprecision(Float32, g)
+end
```
Comment (juliohm) [inline test/runtests.jl]: let's move all helper functions to a test/utils.jl

File: test/primitives.jl [hunk 5]
```diff
@@ -429,6 +452,10 @@
     @test maximum(b) == point(1, 1, 1)
     @test extrema(b) == (point(0, 0, 0), point(1, 1, 1))
 
+    b = Box(point(0, 0), point(1, 1))
+    equaltest(b)
+    isapproxtext(b)
```
Comment (eliascarv) [inline test/primitives.jl]:
```suggestion
    isapproxtest(b)
```
Typo

File: src/multigeoms.jl (hunk 1 above)
Comment (eliascarv) [reply src/multigeoms.jl:57]: Yes, it's:
```
julia> test1(m₁::Multi, m₂::Multi) = m₁.geoms == m₂.geoms
test1 (generic function with 1 method)

julia> test2(m₁::Multi, m₂::Multi) = length(m₁.geoms) == length(m₂.geoms) && all(g -> g[1] == g[2], zip(m₁.geoms, m₂.geoms))
test2 (generic function with 1 method)

julia> t = Triangle((1, 2, 3), (4, 5, 6), (7, 8, 9));

julia> m = Multi([t, t, t, t, t, t, t, t, t]);

julia> using BenchmarkTools

julia> @btime test1(m, m)
  66.956 ns (0 allocations: 0 bytes)
true

julia> @btime test1($m, $m)
  56.977 ns (0 allocations: 0 bytes)
true

julia> @btime test2($m, $m)
  59.417 ns (0 allocations: 0 bytes)
true
```

File: src/primitives/point.jl (hunk 2 above)
Comment (eliascarv) [reply src/primitives/point.jl:52]: Yes, it is:
```
julia> using Meshes

julia> p = Point(1, 2)
Point with Cartesian{NoDatum} coordinates
├─ x: 1.0 m
└─ y: 2.0 m

julia> test1(A::Point, B::Point) = A.coords == B.coords
test1 (generic function with 1 method)

julia> test2(A::Point, B::Point) = to(A) == to(B)
test2 (generic function with 1 method)

julia> using BenchmarkTools

julia> @btime test1($p, $p)
  2.114 ns (0 allocations: 0 bytes)
true

julia> @btime test2($p, $p)
  2.121 ns (0 allocations: 0 bytes)
true
```

**Review by juliohm — APPROVED** [review]

**Pull request merged**